src/PrestaShopBundle/Translation/Exception/InvalidLegacyTranslationKeyException.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Translation\Exception;
+
+/**
+ * Thrown when an invalid key is found in a legacy translation file
+ */
+class InvalidLegacyTranslationKeyException extends \Exception
+{
+    /**
+     * @var string The invalid key
+     */
+    private $key = '';
```

Comment:
Basically the purpose for this exception is to make sure that the all keys stored in the translation file have are well-formatted (see `LegacyFileLoader`). That's why I store the key (so that you can know which one it is) and no the missingElement which exists for debugging purposes only.

I sincerely don't remember the original reason why I wanted to retrieve the key, as in the end I never used it...